presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -148,7 +148,7 @@ public void checkCanShowTablesMetadata(ConnectorTransactionHandle transactionHan
     @Override
     public Set<SchemaTableName> filterTables(ConnectorTransactionHandle transactionHandle, Identity identity, Set<SchemaTableName> tableNames)
     {
-        return tableNames;
+        return tableNames.stream().filter(tableName -> checkTablePermission(transactionHandle, identity, tableName, SELECT)).collect(Collectors.toSet());
```

Comment:
Probably any one of the permissions, not just SELECT -- you could have just INSERT privileges, for example.